projects/destinations_similarity/destinations_similarity/tasks.py

```diff
@@ -0,0 +1,262 @@
+    translate_description_series, preprocess_text)
+from destinations_similarity.processing.feature_engineering import (
+    TextVectorizer)
+
+
+# Logging config
+LOGGER = logging.getLogger(__name__)
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="[%(asctime)s] %(name)s: %(levelname)s | %(message)s"
+)
+
+# Flyte configuration
+LIGHT_RESOURCES = Resources(cpu="0.5", mem="1Gi")
+BASE_RESOURCES = Resources(cpu="1", mem="2Gi")
+INTENSIVE_RESOURCES = Resources(cpu="2", mem="16Gi")
+
+
+@task(retries=3, requests=LIGHT_RESOURCES)
+def get_base_data(generate_city_id: bool) -> pd.DataFrame:
+    """Retrieve base data for the dataset.
+
+    Args:
+        generate_city_id (bool): Informs if an ID must be generated for each
+            row of the dataset.
+
+    Returns:
+        pd.DataFrame: Base dataset.
+    """
+    return get_brazilian_cities_data(
+        get_dataframe, generate_city_id=generate_city_id)
+
+
+@task(retries=3, requests=LIGHT_RESOURCES)
+def scrap_wiki(
+    base_data: pd.DataFrame, wiki: str, lang: str, summary: bool,
+    sections: List[str], sections_tags: Dict[str, List[str]],
+    sections_types: Dict[str, str]
+) -> pd.DataFrame:
+    """Scrap a Wikimedia page for info.
+
+    Args:
+        base_data (pd.DataFrame): Base dataset.
+        wiki (str): Type of wiki ('wikipedia', 'wikivoyage').
+        lang (str): Language of wiki.
+        summary (bool): If the summary must be retrieved.
+        sections (List[str]): Which sections must be retrieved.
+        sections_tags (Dict[str, List[str]]): Which HTML tags must be preserved
+            for a given section.
+        sections_types (Dict[str, str]): How each section will be
+            saved on the dataset, 'str' or 'list'.
+
+    Returns:
+        pd.DataFrame: The updated dataset.
+    """
+    # Initialize scraper
+    extractor = WikiExtractor(wiki=wiki, lang=lang)
+
+    # Setup fields for the sections
+    sections_fields = {
+        section: f"{section}_{wiki}_{lang}".lower().replace(' ', '_')
+        for section in ['summary'] + sections
+    }
+
+    # Initialize dataset
+    dataset = base_data.copy()
+    dataset[f"images_{wiki}_{lang}"] = [[] for _ in range(len(dataset))]
+    for section, field in sections_fields.items():
+        dataset[field] = (
+            [[] for _ in range(len(dataset))]
+            if sections_types.get(section) == 'list' else ""
+        )
+
+    # Retrieve data for each city
+    for i, row in dataset.iterrows():
+        page_name = row[f"title_{wiki}_{lang}"]
+
+        # Set content
+        page_content = extractor.extract_content(
+            page_name, summary=summary, sections=sections,
+            sections_tags=sections_tags, section_types=sections_types
+        )
+        for section, text in page_content.items():
+            dataset.at[i, sections_fields[section]] = text
+
+        # Set images links
+        page_images = extractor.extract_images(page_name)
```

Comment:
Why are we trying to extract image links?

Reply:
This is for future works, in case we want to use the images in the embedding process. We ended up not using them, for now, to avoid making a too complex system.